src/services/download/mod.rs

```diff
@@ -84,6 +91,26 @@ async fn dispatch_download(
     }
 }
 
+/// Dispatches listing files one the appropriate source type.
+async fn dispatch_list_files(
+    source: SourceConfig,
+    filetypes: &'static [FileType],
+    object_id: ObjectId,
+) -> Result<Vec<SourceFileId>, DownloadError> {
+    match source {
+        SourceConfig::Sentry(cfg) => {
+            let vec = sentry::list_files(cfg, filetypes, object_id)
+                .compat()
+                .await?;
+            Ok(vec)
```

Comment:
Can you just directly return the expression here?

```rust
        SourceConfig::Sentry(cfg) => {
            sentry::list_files(cfg, filetypes, object_id)
                .compat()
                .await
        }
```

Reply:
I tend to avoid `Ok()`-wrapping around multi-line expressions as I don't find it very clear.  But sure, maybe the ok-wrapping can be on the returned thing instead.

Reply:
Unless I'm missing something, there's no need for `Ok()` wrapping. See my code example. You already get a `Result<Vec<_>, DownloadError>` from `.await`, which you can directly return.

Reply:
oh, oops.  yes, you're right